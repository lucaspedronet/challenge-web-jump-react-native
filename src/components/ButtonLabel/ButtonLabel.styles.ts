import {RFValue} from 'react-native-responsive-fontsize';
import styled from 'styled-components/native';

export const Container = styled.TouchableOpacity`
  align-items: center;
  justify-content: center;

  padding: ${RFValue(12)}px ${RFValue(28)}px ${RFValue(14)}px;
  width: 100%;
  height: ${RFValue(48)}px;

  border-radius: ${({theme}) =>
    RFValue(theme.size.borderRadius.default + 14)}px;

  background-color: ${({theme}) => theme.colors.button.primary.Enabled};
`;
