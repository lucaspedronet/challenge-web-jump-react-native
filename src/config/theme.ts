const desingSystem = {
  purple: {
    Dark: '#4F26C9',
    Default: '#5F2EE2',
    Lighter: '#F1F0FC',
  },
  pink: {
    Dark: '#D90191',
    Default: '#F501AB',
    Lighter: '#FFF2FB',
  },
  gradients: {
    primary: {
      color_01: '#5F2EE2',
      color_02: '#F501AB',
    },
    secondary: {
      color_01: '#F1F0FC',
      color_02: '#FFF2FB',
    },
  },
  darkNeutrals: {
    Darker: '#12121A',
    Dark: '#1D1D26',
    Default: '#292933',
    Ligth: '#3C3C47',
    Ligther: '#595966',
  },
  lightNeutrals: {
    darker: '#ADADB8',
    light: '#E6E6EB',
    white: '#FFFFFF',
  },
  uiColors: {
    success: {
      default: '#6BE784',
      lighter: '#F0FCF2',
    },
    info: {
      default: '#61DDEC',
      lighter: '#F2FBFC',
    },
    warning: {
      default: '#FFDF66',
      lighter: '#FFFCF2',
    },
    danger: {
      default: '#F9695E',
      lighter: '#FFF5F5',
    },
  },
};

const typography = {
  fontFmaily: {
    primary: { // exemplos de fontes
      Regular: 'Fonte-Regular',
      Medium: 'Fonte-Medium',
      SemiBold: 'Fonte-SemiBold',
      Bold: 'Fonte-Bold',
    },
  },
  fontWeight: {
    Bold: '700',
    SemiBold: '600',
    Medium: '500',
    Regular: '400',
  },
  lineHeight: {
    Medium: 1.2, // Indicada para títulos
    Distant: 1.5, // A altura de linha Distant é a mais indicada para parágrafos.
  },
  fontScale: {
    tagline: 10,
    caption: 11,
    xxxs: 12,
    xxs: 14,
    xs: 16,
    sm: 18,
    md: 20,
    lg: 24,
    xl: 28,
    xxl: 32,
    xxxl: 36,
    display: 40,
  },
};

export const theme = {
  colors: {
    app: {
      black: desingSystem.darkNeutrals.Darker,
      white: desingSystem.lightNeutrals.white,
      primary: desingSystem.pink.Default,
      secodary: desingSystem.purple.Default,
    },
    statusBar: {
      default: desingSystem.darkNeutrals.Darker,
    },
    screens: {
      background: {
        default: desingSystem.darkNeutrals.Darker,
      },
    },
    button: {
      primary: {
        Enabled: desingSystem.purple.Default,
        Disabled: desingSystem.darkNeutrals.Dark,
      },
    },
    title: {
      white: desingSystem.lightNeutrals.white,
    },
    paragraph: {
      default: desingSystem.lightNeutrals.darker,
      white: desingSystem.lightNeutrals.white,
      danger: desingSystem.uiColors.danger,
      success: desingSystem.uiColors.success.default,
      placeholder: desingSystem.darkNeutrals.Ligther,
    },
    label: {
      Enabled: desingSystem.lightNeutrals.white,
      Disabled: desingSystem.darkNeutrals.Ligth,
    },
    icon: {
      default: 'rgba(255, 255, 255, 1)',
      primary: 'rgba(94, 221, 95, 1)',
      successLight: 'rgba(94, 221, 95, 0.1)',
      secondary: 'rgba(43, 47, 67, 1)',
      gray: 'rgba(211, 218, 228, 1)',
      grayLight: 'rgba(245, 246, 248, 1)',
    },
    tabBar: {
      icon: {
        active: desingSystem.lightNeutrals.white,
        inactive: desingSystem.lightNeutrals.darker,
      },
      badge: desingSystem.pink.Default,
    },
  },
  size: {
    card: {
      height: {
        small: 40,
        default: 120,
        large: 145,
        zero: 0,
      },
      width: {
        default: 100,
        medium: '50%',
        zero: '0',
      },
    },
    padding: {
      default: 20,
      large: 24,
      small: 16,
      zero: 0,
    },
    margin: {
      default: 20,
      small: 16,
      zero: 0,
    },
    scaleFont: {
      title: {
        small: typography.fontScale.md,
        default: typography.fontScale.lg,
      },
      paragraph: {
        small: typography.fontScale.xxxs,
        default: typography.fontScale.xxs,
        large: typography.fontScale.xs,
      },
      label: {
        default: typography.fontScale.sm,
      },
    },
    borderRadius: {
      default: 10,
      large: 24,
      medium: 2,
      zero: 0,
    },
    button: {
      small: 60,
      smallLarge: 100,
      default: 180,
    },
    styles: {
      letterSpacing: 2,
      lineHeight: 1.2,
    },
  },
  font: {
    ...typography,
    title: {
      small: typography.fontScale.sm,
      default: typography.fontScale.lg,
    },
  },
};
