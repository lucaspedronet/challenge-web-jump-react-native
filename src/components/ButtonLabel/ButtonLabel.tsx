import React from 'react';
import {StyleSheet, TouchableOpacityProps} from 'react-native';
import { Paragraph } from '../Paragraph/Paragraph';

import * as S from './ButtonLabel.styles';

export interface ButtonLabelProps extends TouchableOpacityProps {
  children: string;
}

export function ButtonLabel({children, ...rest}: ButtonLabelProps) {
  return (
    <S.Container {...rest}>
      <Paragraph
        weight="Medium"
        size="small"
        style={styles.label}>
          {children}
        </Paragraph>
    </S.Container>
  );
}

const styles = StyleSheet.create({
  label: {textAlign: 'center'},
});
