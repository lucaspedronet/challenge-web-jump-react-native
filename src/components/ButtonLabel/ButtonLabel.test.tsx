import React from 'react';
import { render } from '@testing-library/react-native';
import { ButtonLabel, ButtonLabelProps } from './ButtonLabel';
import { ThemeProvider } from 'styled-components/native';
import { theme } from '~config/theme';

const Providers: React.FC = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const wrapper = (props: ButtonLabelProps) => render(
  <ButtonLabel {...props} />,
  {wrapper: Providers}
);

test('renderização padrão', () => {
  const component = wrapper({children: 'Entrar'})
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});
