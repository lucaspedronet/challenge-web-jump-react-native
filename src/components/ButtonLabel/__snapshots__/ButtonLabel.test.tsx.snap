// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderização padrão 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#5F2EE2",
      "borderRadius": 47,
      "height": 94,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 27,
      "paddingLeft": 55,
      "paddingRight": 55,
      "paddingTop": 24,
      "width": "100%",
    }
  }
>
  <Text
    size="small"
    style={
      [
        {
          "color": "#ADADB8",
          "fontSize": 24,
          "fontStyle": "normal",
          "fontWeight": "500",
          "lineHeight": 35,
        },
        {
          "textAlign": "center",
        },
      ]
    }
    weight="Medium"
  >
    Entrar
  </Text>
</View>
`;
